extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/InfinispanClientProcessor.java

```diff
@@ -149,6 +151,9 @@ InfinispanPropertiesBuildItem setup(ApplicationArchivesBuildItem applicationArch
 
         // This is required for netty to work properly
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, "io.netty.channel.socket.nio.NioSocketChannel"));
+        substrateConfig.produce(SubstrateConfigBuildItem.builder()
```

Comment:
Found issue with Infinispan?

Reply:
Yes, the reason is the added RuntimeInitializedClass of `io.netty.channel.epoll.Epoll` in the NettyProcessor [1]. Although the `TransportHelper` is being substituted in [SubstituteTransportHelper](https://github.com/quarkusio/quarkus/blob/master/extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/graal/SubstituteTransportHelper.java) where there isn't a dependency on Epoll, the original [TransportHelper](https://github.com/infinispan/infinispan/blob/master/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/TransportHelper.java) does have this dependency. When building a native image I got an exception that TransportHelper was dependent on Epoll which was runtime initialized. It was a separated commit, but I squashed in the rebase. Although when running it again today without this addition, it didn't seem necessary anymore.

[1] I thought it would be best to put it in the NettyProcessor, then to put it in the ArtemisProcessor. I don't know if there is an convention, some libraries tend to put their Netty code in their own library. For example the InfinispanClientProcessor has `SystemPropertyBuildItem("io.netty.noUnsafe", "true")`, but so does the NettyProcessor.

Reply:
I forgot if there is a best practice @stuartwdouglas 
Ideally if that's "statically" shared better put it in the based extension.

Reply:
I tested again without this. It works on macos (before rebase it didn't), but still not Linux.
```
com.oracle.svm.core.util.UserError$UserException: Class that is marked for delaying initialization to run time got initialized during image building: io.netty.channel.epoll.Epoll
```
or 
```
com.oracle.svm.core.util.UserError$UserException: Class that is marked for delaying initialization to run time got initialized during image building: io.netty.channel.unix.Errors
```